linux_os/guide/system/software/sudo/sudo_remove_nopasswd/bash/shared.sh

```diff
@@ -0,0 +1,3 @@
+# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_debian,multi_platform_ubuntu,multi_platform_ol
+
+sed -i '/^#/!s/\(.*\)[Nn][Oo][Pp][Aa][Ss][Ss][Ww][Dd][:][ ]\(.*\)/\1\2/g' /etc/sudoers /etc/sudoers.d/*
```

Comment:
`sed -i '/^#/!s/\(.*\)!nopasswd\(.*\)/\1\2/gI' /etc/sudoers /etc/sudoers.d/*` is simpler to understand

Reply:
Conversely, I have consulted the man page, and wasn't able to find any notion of lowercase nopasswd, so I make this suggestion:

```suggestion
sed -i '/^#/!s/\(.*\)NOPASSWD:\(.*\)/\1\2/g' /etc/sudoers /etc/sudoers.d/*
```
It differs from the @redhatrises suggestion by 

- including the colon in the pattern, so it is removed along the keyword,
- not including the `!` in the pattern, as I don't think that it is expected,
- and by preserving case sensitivity.